ovve_ui/utils/comms_link.py

```diff
@@ -0,0 +1,278 @@
+                params_dict['resp_rate_meas'] = in_pkt['respiratory_rate_measured']
+                params_dict['resp_rate_set'] = in_pkt['respiratory_rate_set']
+                params_dict['tv_meas'] = in_pkt['tidal_volume_measured']
+                params_dict['tv_set'] = in_pkt['tidal_volume_set']
+                params_dict['ie_ratio_meas'] = in_pkt['ie_ratio_measured']
+                params_dict['ie_ratio_set'] = in_pkt['ie_ratio_set']
+                params_dict['peep'] = in_pkt['peep_value_measured']
+                params_dict['ppeak'] = in_pkt['peak_pressure_measured']
+                params_dict['pplat'] = in_pkt['plateau_value_measurement']
+                params_dict['pressure'] = in_pkt['pressure_measured']
+                params_dict['flow'] = in_pkt['flow_measured']
+                params_dict['tv_insp'] = in_pkt['volume_in_measured']
+                params_dict['tv_exp'] = in_pkt['volume_out_measured']
+                params_dict['tv_rate'] = in_pkt['volume_rate_measured']
+                # Not sure what is tv_min keeping it like in sumulator for now
+                params_dict['tv_min'] = random.randrange(475, 575)
+                params_dict['control_state'] = 0
+                params_dict['battery_level'] = in_pkt['battery_level']
+                # is this class variable necessary?
+                # self. seqnum = params_dict['seq_num']
+                self.comms_adapter.update_params(params_dict)
+                # we got here
+                # print ("UI parameters updated")
+
+            #
+            #time.sleep(1)
+
+    def init_serial(self):
+    
+        #global ser          #Must be declared in Each Function
+        self.ser = serial.Serial()
+        self.ser.baudrate = self.BAUD
+        self.ser.port = self.PORT 
+        #a little less than polling spped from arduino
+        self.ser.timeout = self.SER_TIMEOUT
+        self.ser.open()          #Opens SerialPort
+
+        # print port open or closed
+        if self.ser.isOpen():
+            print ('Open: ' + self.ser.portstr)
+        sleep(1)
+
+    def crccitt(self, hex_string):
+        byte_seq = binascii.unhexlify(hex_string)
+        crc = crc16.crc16xmodem(byte_seq, 0xffff)
+        return '{:04X}'.format(crc & 0xffff)
+
+    def read_all(self, port, chunk_size=200):
+        """Read all characters on the serial port and return them."""
+        if not port.timeout:
+            raise TypeError('Port needs to have a timeout set!')
+
+        read_buffer = b''
+
+        while True:
+            # Read in chunks. Each chunk will wait as long as specified by
+            # timeout. Increase chunk_size to fail quicker
+            byte_chunk = port.read(size=chunk_size)
+            read_buffer += byte_chunk
+            if not len(byte_chunk) == chunk_size:
+                break
+
+        return read_buffer
+
+
+    def start(self) -> None:
+        t = Thread(target=self.process_SerialData, args=())
```

Comment:
You should set self.done = False here in case you stop and restart.  I added this in a later revision of CommSimulator.